### ISSUE TITLE
torch.nn.AdaptiveAvgPool2d didn't return result after 2 minutes running

### ISSUE THREAD

**Issue by cheyennee** [pytorch/pytorch#83062]
### 🐛 Describe the bug

In this extreme example, after 2 mins running, torch.nn.AdaptiveAvgPool2d didn't return any result or throw exception.

```
import torch
results={}
arg_1_0 = 576460752303423488
arg_1_1 = -274877906944
arg_1 = [arg_1_0,arg_1_1,]
arg_class = torch.nn.AdaptiveAvgPool2d(arg_1,)
arg_2_0 = torch.rand([16, 960, 4, 4], dtype=torch.float32)
arg_2 = [arg_2_0,]
results['res'] = arg_class(*arg_2)
```

### Versions

pytorch: 1.8.1
os: windows11
python: 3.8.3

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by redmagnos** [comment]
This seems to have been fixed at least as of version 1.12.0:

```
RuntimeError: adaptive_avg_pool2d: elements of output_size must be greater than or equal to 0 but received {576460752303423488, -274877906944}
```

Making the second argument positive also returns an error:
```
RuntimeError: numel: integer multiplication overflow
```

**Comment by albanD** [comment]
Thanks for checking @redmagnos 
Closing as already fixed
